src/stdatamodels/jwst/transforms/models.py

```diff
@@ -1659,7 +1527,12 @@ def _trace_quadratic(t, x0, y0, model):
             y0 = y0[0].flatten()
             wavelength = wavelength[:, 0].flatten()
 
-        wave_grid = trace_function(t0, x0, y0)
+        trace_function = partial(_poly_with_spatial_dependence, model=model)
+
+        # Create a grid of t0, x0, and y0 values
+        tt, yy = np.meshgrid(t0, y0, indexing="ij")
+        xx = np.meshgrid(t0, x0, indexing="ij")[1]
+        wave_grid = trace_function(tt, xx, yy)
```

Comment:
This change basically makes it so that the inputs all have the same shape as the eventual output before attempting to eval the trace function, rather than putting matrix multiplication into the trace function itself.  I did check the memory usage effects of this for my test NIRCam dataset, and the memory footprint of `wfss_contam` was nearly identical